gateway/managers/syncman/store_kube.go

```diff
@@ -43,123 +43,55 @@ func NewKubeStore(clusterID string) (*KubeStore, error) {
+		return "", "", nil
 	}
-
-	if err := json.Unmarshal([]byte(clusterJSONString), clusters[0]); err != nil {
-		_ = helpers.Logger.LogError(helpers.GetRequestID(context.TODO()), "Unable to unmarshal config map data while watching kube store admin config", nil, nil)
-		return
-	}
-	if clusters[0].ClusterConfig == nil {
-		clusters[0].ClusterConfig = getDefaultAdminConfig().ClusterConfig
-	}
-}
-
-func onAddOrUpdateProjects(obj interface{}, projectMap map[string]*config.Project) map[string]*config.Project {
-	configMap := obj.(*v1.ConfigMap)
-	projectJSONString, ok := configMap.Data["project"]
+	dataJSONString, ok := configMap.Data["data"]
 	if !ok {
-		_ = helpers.Logger.LogError(helpers.GetRequestID(context.TODO()), "Unable to watch kube store project as field (cluster) not found in config map", nil, nil)
-		return nil
+		_ = helpers.Logger.LogError(helpers.GetRequestID(context.TODO()), fmt.Sprintf("%s event occured on resource config map, but (resource) field was not found in config map data", eventType), nil, nil)
+		return "", "", nil
 	}
 
-	v := new(config.Project)
-	if err := json.Unmarshal([]byte(projectJSONString), v); err != nil {
-		_ = helpers.Logger.LogError(helpers.GetRequestID(context.TODO()), "Unable to unmarshal config map data while watching kube store project", nil, nil)
-		return nil
+	var v map[string]interface{}
+	if err := json.Unmarshal([]byte(dataJSONString), &v); err != nil {
+		_ = helpers.Logger.LogError(helpers.GetRequestID(context.TODO()), "Unable to unmarshal resource config map data while watching kube store project", nil, map[string]interface{}{"resourceId": resourceID, "eventType": eventType})
+		return "", "", nil
 	}
-	projectMap[v.ID] = v
-	return projectMap
+	return eventType, resourceID, v
 }
 
-// WatchAdminConfig maintains consistency over all projects
-func (s *KubeStore) WatchAdminConfig(cb func(clusters []*config.Admin)) error {
+// WatchResources maintains consistency over all projects
+func (s *KubeStore) WatchResources(cb func(eventType, resourceID string, resource interface{})) error {
 	go func() {
 		var options internalinterfaces.TweakListOptionsFunc = func(options *v12.ListOptions) {
-			options.LabelSelector = fmt.Sprintf("adminConfig=adminConfig,clusterId=%s", s.clusterID)
+			options.LabelSelector = fmt.Sprintf("clusterId=%s", s.clusterID)
 		}
 		informer := informers.NewSharedInformerFactoryWithOptions(s.kube, 0, informers.WithTweakListOptions(options)).Core().V1().ConfigMaps().Informer()
 		stopper := make(chan struct{})
 		defer close(stopper)
 		defer runtime.HandleCrash() // handles a crash & logs an error
 
-		clusters := []*config.Admin{getDefaultAdminConfig()}
-
 		informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
 			AddFunc: func(obj interface{}) {
-				onAddOrUpdateAdminConfig(obj, clusters)
-				cb(clusters)
+				cb(onAddOrUpdateResource(config.ResourceAddEvent, obj))
```

Comment:
Call the callback only when there is a change

Reply:
Isn't this the responsibility of Kubernetes to throw events only on changes? and if there is a change then why should we check for change

Reply:
Your function throws an empty event if an error occurs internally. You are returning empty values in several cases